pkg/build/builder/source.go

```diff
@@ -127,18 +151,33 @@ func extractInputBinary(in io.Reader, source *api.BinaryBuildSource, dir string)
 	return nil
 }
 
+func hasProxy(source *api.GitBuildSource) bool {
+	return !(len(source.HTTPProxy) == 0 && len(source.HTTPSProxy) == 0)
+}
+
 func extractGitSource(git git.Git, gitSource *api.GitBuildSource, revision *api.SourceRevision, dir string, timeout time.Duration) (bool, error) {
 	if gitSource == nil {
 		return false, nil
 	}
 
-	// Set the HTTP and HTTPS proxies to be used by git clone.
-	originalProxies := setHTTPProxy(gitSource.HTTPProxy, gitSource.HTTPSProxy)
-	defer resetHTTPProxy(originalProxies)
+	if hasProxy(gitSource) {
+		// Set the HTTP and HTTPS proxies to be used by git clone.
+		originalProxies := setHTTPProxy(gitSource.HTTPProxy, gitSource.HTTPSProxy)
+		defer resetHTTPProxy(originalProxies)
+
+		// When the protocol is git:// the HTTPS_PROXY and HTTP_PROXY variables are
+		// ignored. This command will replace git:// with https:// in that case.
+		if strings.HasPrefix(gitSource.URI, "git://") {
```

Comment:
this is wrong... the `git://` protocol is not http/https-based, and should not be using http_proxy/https_proxy. It's over a completely different port, so switching to `https://` could actually break the check (https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#The-Git-Protocol)